dali/operators/reader/loader/video/frames_decoder_gpu.cc

```diff
@@ -557,35 +578,76 @@ bool FramesDecoderGpu::ReadNextFrameWithoutIndex(uint8_t *data, bool copy_to_out
   current_copy_to_output_ = copy_to_output;
   current_frame_output_ = data;
 
+  int frame_to_return_index = -1;
+
+  // Handle the case, when packet has more frames that we have empty spots
+  // in the buffer.
+  // If so, we need to return frame from the buffer before sending last packet.
+  if (frame_index_if_no_pts_ != 0) {
+    if (NumEmptySpots() < (piped_pts_.size())) {
+      for (size_t i = 0; i < frame_buffer_.size(); ++i) {
```

Comment:
Is it possible that we won't find desired frame in the `frame_buffer_` as it has not been decoded yet?

Reply:
I think that theoretically, this can happen. The ultimately resilient solution for this issue I have found is to allocate spaces in the buffer dynamically as much as needed.
In practice, I have never encountered this issue yet, we have buffer big enough for the files that we have seen so far.  

I would like to leave it as it is now. When we do decoupling mentioned above, we can revisit it and allow for buffer dynamic extension or something like that. Is that ok?

Reply:
Maybe we can error in such case instead of silently returning the wrong result so the user can provide us with a failing sample and we can adjust?

Reply:
I think right now it should error out with `Could not find empty slot in the frame buffer`. It should try to find the next frame, fail to do it, than call for another frame, try to place it in the buffer and error out due to lack of empty spots.